### ISSUE TITLE
mesa build fails with llvm 10: undefined reference to `getPollyPluginInfo()'

### ISSUE THREAD

**Issue by tjaalton** [llvm/llvm-project#191]
Hi,
Building Mesa 20.0.x with llvm 10 fails with something that appears to be a bug in llvm itself:

/usr/bin/ld: /usr/lib/llvm-10/lib/libclangCodeGen.a(BackendUtil.cpp.o): in function `(anonymous namespace)::EmitAssemblyHelper::EmitAssemblyWithNewPassManager(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >)':
(.text._ZN12_GLOBAL__N_118EmitAssemblyHelper30EmitAssemblyWithNewPassManagerEN5clang13BackendActionESt10unique_ptrIN4llvm17raw_pwrite_streamESt14default_deleteIS5_EE+0x1f15): undefined reference to `getPollyPluginInfo()'
collect2: error: ld returned 1 exit status

I've tried with rc3/rc4 and rc5 plus https://reviews.llvm.org/D74464

**Comment by serge-sans-paille** [comment]
Hi and thanks for reporting.
How was the llvm 10 you're using configured ?

**Comment by tjaalton** [comment]
It's from Debian, and the full debian/rules file that shows the configuration (though maybe hard to parse) can be found here:
https://salsa.debian.org/pkg-llvm-team/llvm-toolchain/-/blob/10/debian/rules

the full build log of rc5 can be found here
https://buildd.debian.org/status/fetch.php?pkg=llvm-toolchain-10&arch=amd64&ver=1%3A10.0.0%7E%2Brc5-1&stamp=1584671213&raw=0

**Comment by tjaalton** [comment]
FWIW, Fedora does not seem to suffer from this, but they don't enable polly either from what I can tell..

**Comment by sylvestre** [comment]
@tjaalton we have a pending patch (a cherry pick from @serge-sans-paille patch)

See here:
https://bugs.llvm.org/show_bug.cgi?id=44870
Closing this one as it is a dup of bugzilla's

**Comment by tjaalton** [comment]
I know, since I tested the version with that patch ("rc5 plus https://reviews.llvm.org/D74464" as provided by Gianfranco) and it didn't help
